File: Pythonsolutions/eu37.py
if __name__=="__main__":
	maxnbr = 1000000
	sum = 0
	primes = []
	i = 0
	while(i<maxnbr):
		primes.append(True)
		i+=1
	primes[0] = False
	primes[1] = False
	i = 2
	while(i<maxnbr):
		n = i+i
		while(n<maxnbr):
			primes[n] = False
			n += i
		i+=1	
	for q in range(8,maxnbr):
		if(primes[q]):
			s1 = str(q)
			s2 = s1
			a = True
			while(s1!=''):
				if(not primes[int(s1)]):
					a = False
					break
				s1 = s1[1:]
			b = True
			while(s2!=''):
				if(not primes[int(s2)]):
					b = False
					break
				s2 = s2[:len(s2)-1]
			if(a and b):
				sum += q
	print 'resulting sum is: ',sum

File: Perlsolutions/eu37.pl
#!/usr/bin/perl
use strict;
use warnings;

my $max_nbr = 1000000;
my $sum = 0;
my @primes;
for my $n (2 .. $max_nbr){
	push @primes, 1;
}
$primes[0] = 0;
$primes[1] = 0;
my $i = 2;
while($i < $max_nbr){
	my $p = 2*$i;
	while($p<$max_nbr){
		$primes[$p] = 0;
		$p += $i;
	}
	$i++;
}
for my $q (8 .. $max_nbr){
	if($primes[$q]){
		my $s1 = $q.'';
		my $s2 = $s1;
		my $check1 = 1;
		while($s1 ne ''){
			if(not $primes[$s1+0]){
				$check1 = 0;
				last;
			}
			my $stop = length $s1;
			$s1 = substr $s1,1,$stop;
		}
		my $check2 = 1;
		while($s2 ne ''){
			if(not $primes[$s2+0]){
				$check2 = 0;
				last;
			}
			my $stop = length $s2;
			$s2 = substr $s2,0,$stop-1;
		}
		if($check1 && $check2){
			$sum += $q;
		}
	}
}
print "resulting sum is: $sum";